syntax = "proto3";

package quickbook;

service CarBooker {
    rpc AddInventory(AddInventoryRequest) returns (AddInventoryReply) {}
    rpc BookCar(BookCarRequest) returns (BookCarReply) {}
}

message Car {
    uint32 id = 1;
    string brand = 2;
    string model = 3;
    uint32 quantity = 4;
}

message AddInventoryRequest {
    uint32 car_id = 1;
    uint32 quantity = 2;
}

message AddInventoryReply {
    Car car = 1;
}

message BookCarRequest {
    uint32 car_id = 1;
    uint32 quantity = 2;
}

message BookCarReply {
    Car car = 1;
}
